### PR TITLE
Added IKillable interface for HP + Added player & enemy deaths

### PR DESCRIPTION
**Pull request by redknight990**
- IKillable interface for game objects with hit points
- Player & Enemy deaths w/ animations
- Player input disabled when dead
- Fixed upper body mask causing animation bug

### CODE REVIEW

File: Hush/Assets/Scripts/Enemies/Enemy.cs [hunk 1]
```diff
@@ -49,16 +48,24 @@ private enum State
 
     #region Public Variables
 
-    // TODO: Not sure an ID is needed
-    public int ID { get; set; }
-
```
Comment (arejayelle) [inline Hush/Assets/Scripts/Enemies/Enemy.cs:54]: Delicious

File: Hush/Assets/Scripts/Enemies/EnemySword.cs [hunk 2]
```diff
@@ -24,7 +26,7 @@ void OnTriggerEnter(Collider other)
         {
             if (enemy.IsAttacking && other.CompareTag(Tags.Player))
             {
-                // TODO: Damage player
+                other.GetComponent<IKillable>()?.TakeDamage(damage);
```
Comment (arejayelle) [inline Hush/Assets/Scripts/Enemies/EnemySword.cs:29]: lmao javascript ass code

File: Hush/Assets/Scripts/Enemies/IEnemy.cs [hunk 3]
```diff
@@ -1,8 +1,5 @@
...
-{   
-    // TODO: Not sure we need an ID
-    public int ID { get; set; }
-    void Die();
-    void TakeDamage(int amount);
-    void PerformAttack();
+using Common;
+
+public interface IEnemy : IKillable
+{
```
Comment (arejayelle) [inline Hush/Assets/Scripts/Enemies/IEnemy.cs:4]: mm useful

File: Hush/Assets/Scripts/Player/Enums/PlayerAnimator.cs [hunk 4]
```diff
@@ -9,5 +9,16 @@ public static class PlayerAnimator
 		public static int HeavyAttack => Animator.StringToHash("HeavyAttack");
 		public static int SpellAttack => Animator.StringToHash("SpellAttack");
 		public static int SpellSpecialAttack => Animator.StringToHash("SpellSpecialAttack");
+        public static int Dead => Animator.StringToHash("Dead");
+        public static class Layer
+        {
```
Comment (arejayelle) [inline Hush/Assets/Scripts/Player/Enums/PlayerAnimator.cs:14]: Lint

**Review by arejayelle — APPROVED** [review]
All good!

**Review by arejayelle — DISMISSED** [review]
Good stuff

**Review by daniela-venuta — APPROVED** [review]

**Pull request merged**